### PR TITLE
Create class to store the EA components

### PR DESCRIPTION
**Pull request by yuval-ya**
I didn't finish it but you can put ur suggestions (as a comment or commit)

### CODE REVIEW

File: ea-engine/main.py [hunk 1]
```diff
@@ -36,64 +35,16 @@
...
-                                  ngen=400, verbose=False)
+def main():
+    ea = EA(len(data["drivers"]), len(data["orders"]), data) \
+        .set_crossover(Crossovers.SINGLE_POINT) \
+        .set_mutate(Mutates.SHUFFLE, indpb=0.05) \
+        .set_selection(Selection.TOURNAMENT, tournsize=3) \
+        .create_population(100)
 
+    result, log = ea.eval_model()
+    best_individual = tools.selBest(result, k=1)[0]
```
Comment (tamirho) [inline ea-engine/main.py:46]: Let's consider also encapsulating this functionality in the `EA` class as a further task.

File: ea-engine/main.py [hunk 2]
```diff
@@ -36,64 +35,16 @@
...
-
-result, log = algorithms.eaSimple(pop, toolbox,
-                                  cxpb=0.8, mutpb=0.2,
-                                  ngen=400, verbose=False)
+def main():
+    ea = EA(len(data["drivers"]), len(data["orders"]), data) \
+        .set_crossover(Crossovers.SINGLE_POINT) \
+        .set_mutate(Mutates.SHUFFLE, indpb=0.05) \
+        .set_selection(Selection.TOURNAMENT, tournsize=3) \
+        .create_population(100)
```
Comment (tamirho) [inline ea-engine/main.py:43]: like that

File: ea-engine/ea.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        self.__toolbox.register("mutate", mutate_type, **kwargs)
+        return self
+
+    def set_selection(self, selection_type, **kwargs):
+        self.__toolbox.register("select", selection_type, **kwargs)
+        return self
+
+    def create_population(self, num_individuals):
+        self.__pop = self.__toolbox.population(n=num_individuals)
+        return self
```
Comment (tamirho) [inline ea-engine/ea.py:35]: Why this is not a part of `eval_mode`?
Maybe we should change this to `set_population` as all other parameters, and initial the population in the eval mode?

**Review by tamirho — CHANGES_REQUESTED** [review]
It looks really good!
I think we should consider setting default values to parameters and letting the user set them with setters (as you did) but leave the construction part to the `eval` method or add a new `build` method.

**Comment by yuval-ya** [comment]
> 
I get your idea.

**Review by vladib6 — APPROVED** [review]
Hi all, I still trying to learn python syntax.

**Review by tamirho — APPROVED** [review]

**Pull request merged**